docs/tutorial.py

```diff
@@ -0,0 +1,78 @@
+# ruff: noqa
+"""
+The kasa library is fully async and methods that perform IO need to be run inside an async couroutine.
+These examples assume async code is called from within `async def` or are running in a asyncio REPL
+(python -m asyncio)
+
+The main entry point for the api is :func:`~kasa.Discover.discover` and
+:func:`~kasa.Discover.discover_single` which return Device objects.
+Most newer devices require your tplink cloud username and password but this can be ommitted for older devices.
+
+>>> from kasa import Device, Discover
+
+:func:`~kasa.Discover.discover` returns a list of devices on your network
+
+>>> devices = await Discover.discover(username="user@mail.com", password="great_password")
+>>> for dev in devices:
+>>>     await dev.update()
+>>>     print(dev.host)
+127.0.0.1
+127.0.0.2
+
+:func:`~kasa.Discover.discover_single` returns a single device by hostname
+
+>>> dev = await Discover.discover_single("127.0.0.1", username="user@mail.com", password="great_password")
```

Comment:
```suggestion
>>> dev = await Discover.discover_single("127.0.0.1", username="user@example.com", password="great_password")
```
Better use known example addresses, just in case :-)

Reply:
Done